vulnerabilities/importers/alpine_linux.py

```diff
@@ -20,124 +20,192 @@
-            advisory_directory_page = requests.get(advisory_directory_link).content
-            advisory_directory_page = BeautifulSoup(advisory_directory_page, features="lxml")
+            advisory_directory_page = fetch_response(advisory_directory_link).content
             advisory_links.extend(
-                [
-                    f"{advisory_directory_link}{anchore_tag.text}"
-                    for anchore_tag in advisory_directory_page.find_all("a")
-                    if anchore_tag.text.endswith("yaml")
-                ]
+                fetch_advisory_links(advisory_directory_page, advisory_directory_link)
             )
-
-        return advisory_links
-
-    def updated_advisories(self) -> Set[Advisory]:
-        advisories = []
-        advisory_links = self.fetch_advisory_links()
         for link in advisory_links:
-            advisories.extend(self._process_link(link))
-
-        return self.batch_advisories(advisories)
-
-    def _process_link(self, link) -> List[Advisory]:
-        advisories = []
-        yaml_response = requests.get(link).content
-        record = saneyaml.load(yaml_response)
-
-        if record["packages"] is None:
-            return advisories
-
-        for p in record["packages"]:
-            advisories.extend(
-                self._load_advisories(
-                    p["pkg"],
-                )
-            )
-
+            record = fetch_response(link).json()
+            if not record["packages"]:
+                raise Exception(f'"packages" not found in {link}')
+            advisories.extend(process_record(record))
         return advisories
 
-    @staticmethod
-    def _load_advisories(
-        pkg_infos: Mapping[str, Any],
-    ) -> List[Advisory]:
 
-        advisories = []
-
-        for fixed_vulns in pkg_infos["secfixes"].values():
-
-            if fixed_vulns is None:
-                continue
-
-            for vuln_ids in fixed_vulns:
-                vuln_ids = vuln_ids.split()
-                references = []
-                for reference_id in vuln_ids[1:]:
-
-                    if reference_id.startswith("XSA"):
-                        xsa_id = reference_id.split("-")[-1]
-                        references.append(
-                            Reference(
-                                reference_id=reference_id,
-                                url="https://xenbits.xen.org/xsa/advisory-{}.html".format(xsa_id),
-                            )
-                        )
-
-                    elif reference_id.startswith("ZBX"):
-                        references.append(
-                            Reference(
-                                reference_id=reference_id,
-                                url="https://support.zabbix.com/browse/{}".format(reference_id),
-                            )
+def fetch_response(url):
+    """
+    Fetch and return `response` from the `url`
+    """
+    response = requests.get(url)
+    if response.status_code == 200:
```

Comment:
Are there instances where the response is not 200, if so you'll like to document that (in a comment maybe) or we won't need this function.

Reply:
This is to ensure that we are able to fetch from the string that we are passing as URL, if we are not able to fetch anything from that we should know that and correct our code if needed